### PR TITLE
cmake-utils.eclass: Add QA check that cmake-utils_src_prepare was run

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: eclass/cmake-utils.eclass [hunk 1]
```diff
@@ -441,6 +441,8 @@ cmake-utils_src_prepare() {
 	fi
 
 	popd > /dev/null || die
+
+	CMAKE_UTILS_SRC_PREPARE_HAS_RUN=1
```
Comment (mgorny) [inline eclass/cmake-utils.eclass]: Maybe prefix it with `_` as a private variable.

File: eclass/cmake-utils.eclass [hunk 2]
```diff
@@ -465,6 +467,9 @@ cmake-utils_src_prepare() {
 cmake-utils_src_configure() {
 	debug-print-function ${FUNCNAME} "$@"
 
+	[[ ! ${CMAKE_UTILS_SRC_PREPARE_HAS_RUN} && ${EAPI} != 5 ]] && \
```
Comment (mgorny) [inline eclass/cmake-utils.eclass]: `\` is not necessary when `&&` is last on the line.

File: eclass/cmake-utils.eclass (hunk 1 above)
Comment (SoapGentoo) [inline eclass/cmake-utils.eclass]: I'd like to keep it semi-official for the reason that some packages call cmake in a subproject of some package. For such exceptional cases `CMAKE_UTILS_SRC_PREPARE_HAS_RUN` can be defined globally to disable the QA check.

Comment (mgorny) [inline eclass/cmake-utils.eclass]: If you need packages to disable this, then give them an official explicit option for this purpose. Do not encourage developers to go playing around with internals. They do that enough without your help.

File: eclass/cmake-utils.eclass [hunk 3]
```diff
@@ -441,6 +447,12 @@ cmake-utils_src_prepare() {
 	fi
 
 	popd > /dev/null || die
+
+	# make ${S} read-only in order to detect badly written build-systems
+	[[ ${CMAKE_UTILS_QA_SRC_DIR_READONLY} && ! ${CMAKE_IN_SOURCE_BUILD} ]] &&
```
Comment (kensington) [inline eclass/cmake-utils.eclass]: Please use full ``if`` clause.

File: eclass/cmake-utils.eclass [hunk 4]
```diff
@@ -465,6 +477,9 @@ cmake-utils_src_prepare() {
 cmake-utils_src_configure() {
 	debug-print-function ${FUNCNAME} "$@"
 
+	[[ ! ${_CMAKE_UTILS_SRC_PREPARE_HAS_RUN} ]] &&
```
Comment (kensington) [inline eclass/cmake-utils.eclass]: Please use full ``if`` clause here too.

File: eclass/cmake-utils.eclass [hunk 5]
```diff
@@ -441,6 +447,12 @@ cmake-utils_src_prepare() {
 	fi
 
 	popd > /dev/null || die
+
+	# make ${S} read-only in order to detect badly written build-systems
+	[[ ${CMAKE_UTILS_QA_SRC_DIR_READONLY} && ! ${CMAKE_IN_SOURCE_BUILD} ]] &&
+		chmod -R u-w,g-w,o-w "${S}"
```
Comment (kensington) [inline eclass/cmake-utils.eclass]: Does ``a-w`` do the same thing?

**Pull request closed without merging**